src/main/java/net/neoforged/neoforge/network/registration/NetworkRegistry.java

```diff
@@ -816,7 +817,7 @@ public boolean onVanillaNetworkConnectionEstablished(ClientConfigurationPacketLi
 
         //Negotiation failed. Disconnect the client.
         if (!playNegotiationResult.success()) {
-            sender.getConnection().disconnect(Component.translatableWithFallback("neoforge.network.negotiation.failure.vanilla.client.not_supported", "You are trying to connect to a server that is running NeoForge, but you are not. Please install NeoForge Version: %s to connect to this server.", NeoForgeVersion.getVersion()));
+            sender.getConnection().disconnect(Component.translatableWithFallback("neoforge.network.negotiation.failure.vanilla.server.not_supported", "You are trying to connect to a server that is not running NeoForge, but you are not. A connection could not be established.", NeoForgeVersion.getVersion()));
```

Comment:
Same lang comment here.

Potentially we could reduce the duplication by first negotiating configuration and play, and then check `if (!configurationNegotiationResult.success() || !playNegotiationResult.success())`

Reply:
Depending on the channel count, negotiation could be an expensive procedure to complete. Hence them being split.

I would leave it like this for now, but we can for sure refactor this behaviour in a new version down the line.